Allow ipv6_cidr_block to be assigned to peering_connection

martinssipenko opened the pull request:
I've encountered an issue when trying to use `destination_ipv6_cidr_block` with `vpc_peering_connection_id`. I believe this fixes #14564.

I'm not sure about whether `destination_ipv6_cidr_block` should be allowed together with other types (`nat_gateway_id`, `instance_id`, `network_interface_id`).

Also, I could not find any acceptance test for this functionality, so I've not added any new tests to cover this. I'm happy to add tests, however I might need guidance.

stack72 commented:
Hi @martinssipenko 

Please can you add an acceptance test that covers the new behaviour and also make an addition to the documentation?

Thanks so much

Paul

martinssipenko commented:
Hi @stack72 

I've added acceptance test and successfully ran in locally against my AWS account.

Regarding documentation - not sure what should be added, as by reading docs I assumed that the behavior I've implemented would work. Only thing that comes to mind is adding a note to state the fact that `destination_ipv6_cidr_block` can be used only be used together with `gateway_id`, `egress_only_gateway_id` and `vpc_peering_connection_id`.

But perhaps instead we should implement support for the remaining `instance_id` and `network_interface_id`? Don't think it makes sense adding IPv6 support for `nat_gateway_id`, since no NAT is used for IPv6 (egress only gateway is used instead of NAT for IPv6).

Let me know what you think..

martinssipenko commented:
@stack72  any updates on this?

martinssipenko commented:
Closing this Pull Request in favour of https://github.com/terraform-providers/terraform-provider-aws/pull/879
